src/Core/src/Handlers/Picker/PickerHandler.Android.cs

```diff
@@ -86,25 +84,33 @@ public static void MapVerticalTextAlignment(IPickerHandler handler, IPicker pick
 			handler.PlatformView?.UpdateVerticalAlignment(picker.VerticalTextAlignment);
 		}
 
-		void OnFocusChange(object? sender, global::Android.Views.View.FocusChangeEventArgs e)
+		internal static void MapFocus(IPickerHandler handler, IPicker picker, object? args)
 		{
-			if (PlatformView == null)
-				return;
-
-			if (e.HasFocus)
+			if (handler.IsConnected() && !picker.IsFocused)
 			{
-				if (PlatformView.Clickable)
-					PlatformView.CallOnClick();
-				else
-					OnClick(PlatformView, EventArgs.Empty);
+				if (args is FocusRequest request)
+				{
+					handler.PlatformView.Focus(request);
+				}
+
+				handler.PlatformView.CallOnClick();
 			}
-			else if (_dialog != null)
+		}
+
+		internal static void MapUnfocus(IPickerHandler handler, IPicker picker, object? args)
```

Comment:
Can we call up to ViewHandler.MapUnfocus for this code?
And then we don't need to duplicate any code